src/signals/incident/components/form/ContainerSelect/Selector/WfsLayer/ContainerLayer/ContainerLayer.tsx

```diff
@@ -98,16 +151,48 @@ export const ContainerLayer: FunctionComponent<DataLayerProps> = ({ featureTypes
           layerInstance.addLayer(marker);
         }
       });
+
+      layerInstance.on('clusterclick', /* istanbul ignore next */(event: { layer: ClusterMarker }) => {
+        const { _maxZoom: maxZoom } = layerInstance;
+        if (shouldSpiderfy(event.layer, maxZoom)) {
+          if (selectedCluster.current) {
+            event.layer.spiderfy();
+            const latlng = event.layer.getLatLng();
+            const selectedLatLng = selectedCluster.current.getLatLng();
+            if (isEqual(latlng, selectedLatLng)) selectedCluster.current = undefined;
+            else selectedCluster.current = event.layer;
+          } else {
+            selectedCluster.current = event.layer;
+            selectedCluster.current.spiderfy();
+          }
+        } else {
+          // use this offset when zoomToBounds to keep the markers above the panel in mobile view
+          event.layer.zoomToBounds({ paddingTopLeft: [0, -300] });
+        }
+      });
+
+      /* istanbul ignore next */
+      if (selectedCluster.current) {
+        const selectedLatLng = selectedCluster.current.getLatLng();
+        const cluster = layerInstance.getLayers().find(layer => {
```

Comment:
`(layerInstance.getLayers() as ClusterMarker[])` would remove the need for extra type casting later on